index.js

```diff
@@ -48,18 +47,18 @@ function equal(a, b) {
     if (hasMap && (a instanceof Map) && (b instanceof Map)) {
```

Comment:
```suggestion
    if (hasMap && (a instanceof Map)) {
```

This modification may be made to all similar constructor/instance checks that are made to `a` and `b`. Only one needs to be checked since we already compare `a.constructor === b.constructor` prior to this: https://github.com/FormidableLabs/react-fast-compare/blob/5a9c1ac30964a49c38d31a4c093299f677930c53/index.js#L15

Reply:
Removing these checks gets our size down 7 bytes to `644`. Unfortunately with diff:

```diff
diff --git a/index.js b/index.js
index cafa587..888dcd6 100644
--- a/index.js
+++ b/index.js
@@ -44,7 +44,7 @@ function equal(a, b) {
     //
     //    **Note**: `i` access switches to `i.value`.
     var it;
-    if (hasMap && (a instanceof Map) && (b instanceof Map)) {
+    if (hasMap && (a instanceof Map)) {
       if (a.size !== b.size) return false;
       it = a.entries();
       while (!(i = it.next()).done)
@@ -55,7 +55,7 @@ function equal(a, b) {
       return true;
     }
 
-    if (hasSet && (a instanceof Set) && (b instanceof Set)) {
+    if (hasSet && (a instanceof Set)) {
       if (a.size !== b.size) return false;
       it = a.entries();
       while (!(i = it.next()).done)
```

we get two new test failures in `yarn test` of:

```
  1) basics
       Maps
         map and a pseudo map are not equal:

      AssertionError [ERR_ASSERTION]: Expected values to be strictly equal:

true !== false

      + expected - actual

      -true
      +false
      
      at Context.<anonymous> (test/node/basics.spec.js:25:18)
      at processImmediate (internal/timers.js:439:21)

  2) basics
       Sets
         set and pseudo set are not equal:

      AssertionError [ERR_ASSERTION]: Expected values to be strictly equal:

true !== false

      + expected - actual

      -true
      +false
      
      at Context.<anonymous> (test/node/basics.spec.js:25:18)
      at processImmediate (internal/timers.js:439:21)
```